File: 1051-height-checker/1051-height-checker.cpp
class Solution {
public:
    int heightChecker(vector<int>& heights) {
        int res = 0; 
        std::vector<int> count(101, 0);

        for (int i = 0; i < heights.size(); i++) count[heights[i]]++;

        int i = 1, j = 0; 
        while (i < 101){
			// check value of count
            if(count[i] == 0){
                i++;
            } else {
				// compare the index of count with value of height
                if (i != heights[j]) 
                    res++;
                j++; 
                count[i]--;
            }
        }

        return res;
    }
};

File: 1051-height-checker/1051-height-checker.rs
impl Solution {
    pub fn height_checker(heights: Vec<i32>) -> i32 {
        let mut heights_sort = heights.clone();
        heights_sort.sort_unstable();
        heights
            .iter()
            .zip(heights_sort.iter())
            .filter(|(h, hs)| h != hs)
            .count() as i32
    }
}